src/api/firebase.js

```diff
@@ -73,14 +74,47 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem(checked, listId, itemId) {
+export async function updateItem(
+	checked,
+	listId,
+	itemId,
+	prevDateLastPurchased,
+	prevDateNextPurchased,
+) {
 	const listItemRef = doc(db, listId, itemId);
 	const listItemSnap = await getDoc(listItemRef);
+
+	const dateFirstPurchased = listItemSnap.data().dateCreated;
```

Comment:
Just a little code style thought: I believe all of these different items from `listItemSnap.data()` could be destructed in a single call, ie `const { dateCreated, etc } = listItemSnap.data()`